### DISCUSSION TITLE
Who knows a ESP8266 compatible device ...

### DISCUSSION THREAD

**Discussion by Pewibe** [arendst/Tasmota#11871] [Q&A] [answered]
Who knows a ESP8266 compatible device with these parameters:
- has a (mountable) case - no board only
- has a serial TTL / 3.3V interface (Tx, Rx) at a plugable socket
- has a power supply on board or a connection for a DC power supply

I like to use the Smart Meter Interface of Tasmota together with an IR TTL read/write head.

**Comment by sfromis** [comment, marked as answer]
There are hardly any commercial ESP8266 products not being devices built for specific end user purposes. And the few "more open" I know of would need you to supply a case, which can be bought cheaply. Of course, if you put a board in a case, you can use a regular dev board like NodeMCU.

The Sonoff S20 plug could be a starting point, you could easily solder on an internal header to plug into tx/rx. This plug is very easy to open and work with.

**Reply by Pewibe** [reply]
Thanks for your thoughts @sfromis, but I'm not missing ideas how to modify a device. See this picture:
![image](https://user-images.githubusercontent.com/42893315/115995106-3a082d00-a5da-11eb-9ee5-5bc02aecaaa8.png)
![image](https://user-images.githubusercontent.com/42893315/115995133-51dfb100-a5da-11eb-91b6-43d0b9fb4d8a.png)
I'm looking for a "zero work" device!

**Reply by joba-1** [reply]
M5 atom lite? 

It has m2 hole (mountable).
It has grove plug or standard 0.1in spaced female headers or usb-c for serial.
It has usb-c (and other options) for dc power.
It even has an IR led, a button and an rgb led.

Downside: not esp8266 but esp32. Probably wont stop you :)
